packages/fxa-admin-panel/src/components/AccountSearch/index.tsx

```diff
@@ -49,18 +49,67 @@ export const GET_ACCOUNT_BY_EMAIL = gql`
   }
 `;
 
+// new query for getting account by UID
+export const GET_ACCOUNT_BY_UID = gql`
+  query getAccountByUid($uid: String!) {
+    accountByUid(uid: $uid) {
+      uid
+      createdAt
+      emails {
+        email
+        isVerified
+        isPrimary
+        createdAt
+      }
+      emailBounces {
+        email
+        createdAt
+        bounceType
+        bounceSubType
+      }
+    }
+  }
+`;
+
+function validateUID(uid: string) {
+  // checks if input string is in uid format
+  if (/^[0-9a-fA-F]{32}/.test(uid)) {
+    // alphanumeric, 32 digit
+    return true;
+  }
+  return false;
+}
+
 export const AccountSearch = () => {
   const [inputValue, setInputValue] = useState<string>('');
   const [showResult, setShowResult] = useState<Boolean>(false);
-  const [getAccount, { loading, error, data, refetch }] = useLazyQuery(
-    GET_ACCOUNT_BY_EMAIL
-  );
+  // define two queries to search by either email or uid.
+  const [getAccountbyEmail, emailResults] = useLazyQuery(GET_ACCOUNT_BY_EMAIL);
+  const [getAccountbyUID, UIDResults] = useLazyQuery(GET_ACCOUNT_BY_UID);
+  // choose which query result to show based on type of query made
+  const [isEmail, setIsEmail] = useState<Boolean>(false);
+  const queryResults = isEmail && showResult ? emailResults : UIDResults;
 
   const handleSubmit = (event: React.FormEvent) => {
     event.preventDefault();
-    getAccount({ variables: { email: inputValue } });
-    // Don't hide after the first result is shown
-    setShowResult(true);
+    const isUID = validateUID(inputValue);
+    // choose correct query if email or uid
+    if (isUID && inputValue.search('@') == -1 && inputValue != '') {
```

Comment:
the other checks of `inputValue` are already covered by `validateUID` so we don't need them here